src/editor/Editor.java

```diff
@@ -153,18 +152,19 @@ public void onEvent(Event event) {
     private void editorPlaceTile(EditableTile tile, Layer layer, int x, int y) {
         switch (layer) {
             case BOTH:
-
+                if (tile.preferredLayer == Layer.BOTH || tile.preferredLayer == Layer.UPPER) {
+                    editorPlaceTile(tile, Layer.UPPER, x, y);
+                    if (tile.fullTileWhenBoth || objectToEnum(GameManager.getInstance().getMap().getBottomLayer(x,y)).fullTileWhenBoth)
+                    {
+                        editorPlaceTile(EMPTY, Layer.BOTTOM, x, y);
+                    }
+                }
                 if (tile.preferredLayer == Layer.BOTH || tile.preferredLayer == Layer.BOTTOM) {
-                    this.layer = Layer.BOTTOM;
                     editorPlaceTile(tile, Layer.BOTTOM, x, y);
-                }
-                if (tile.preferredLayer == Layer.BOTH || tile.preferredLayer == Layer.UPPER) {
-                    if (tile == WALL)
+                    if (tile.fullTileWhenBoth || objectToEnum(GameManager.getInstance().getMap().getUpperLayer(x,y)).fullTileWhenBoth)
                     {
-                        tile = EMPTY;
+                        editorPlaceTile(EMPTY, Layer.UPPER, x, y);
```

Comment:
Wydaje mi się, że jeśli będzie tile, który ma preferreeLayer = BOTH i fullTileWhenBoth = true, to postawi się tylko na dolnej warstwie, bo górna zostanie wyczyszczona